HarmonyPatches/SceneTransitionPatch.cs

```diff
@@ -58,6 +58,20 @@ private static void Postfix(MultiplayerLevelScenesTransitionSetupDataSO __instan
                 var randomPresetIndex = random.Next(PluginConfig.Instance.ExtraColorPresetsV2.Count());
                 selectedPreset = PluginConfig.Instance.ExtraColorPresetsV2[randomPresetIndex];
             }
+            else if (PluginConfig.Instance.SelectedPresetId == MinimalExtraColorPreset.randomUniqueItem.colorSchemeId)
+            {
+                Plugin.Log.Info($"Preset selection set to uniquely random, picking from available presets…");
+                var randomPresetIndex = random.Next(Plugin.ExtraColorPresetsUniqueSelectable.Count());
+                selectedPreset = Plugin.ExtraColorPresetsUniqueSelectable[randomPresetIndex];
+                
+                Plugin.ExtraColorPresetsUniqueSelectable.RemoveAt(randomPresetIndex);
```

Comment:
I'm not sure how this approach handles when new presets are added in the config. It might not cause issues when the preset list is sparse and you don't change it often, but personally I would prefer an approach that keeps track of the presets that were selected already, and if that list matches every one of the available presets then it gets reset.

The current way you're handling it does not seem to take into account the possibility for presets to be added/removed during a single play session, but I may be misinterpreting how this works, so feel free to correct me if I'm wrong.

Reply:
The shortcoming of the current approach is that if you delete a preset or add a new one, that cannot be reflected in the already pre-generated list until such time that you exhaust the options in the list, which might be a problem if one of those presets goes missing.
This is why I suggested storing a list of all presets that were selected so far, then iterate over the list of available presets, and add the ones that do not have their ID in the already selected list to the pool of possible presets, and take a random one from that, or generate it fresh if the pool is empty.
If that's unclear I'm happy to elaborate on it further, but basically I want the setting to behave like a blocklist for filters that should not be selected anymore and that would alleviate my concerns, since if a filter on the blocklist gets deleted it will simply be ignored, and if new ones are added, they will not be in the list and therefore eligible to be selected next time.

Reply:
No no it's clear to me, just didn't think of doing this in that way (creating a list and plucking things out until nothing was left seemed more efficient to me). Honestly just forgot to add/remove from the list as presets were added/removed the first time (woops). It was also already re-generating the list if it used up everything within it at the time of selecting the last scheme from the list.